#swervebot: # THIS IS A SPACE NAME 
    # Publish all joint states -----------------------------------
    joint_state_controller:
        type: "joint_state_controller/JointStateController"
        publish_rate: 10

    # Controller for independent wheels --------------------------
    swerve_controller:
        type: "swerve_controller/SwerveController"
        
        # Joints
        lf_wheel: "wheel_fl_joint"
        rf_wheel: "wheel_fr_joint"
        lh_wheel: "wheel_bl_joint"
        rh_wheel: "wheel_br_joint"
        lf_steering: "steering_fl_joint"
        rf_steering: "steering_fr_joint"
        lh_steering: "steering_bl_joint"
        rh_steering: "steering_br_joint"

        # Range of motion of steering motors
        min_steering_angle: -1.58
        max_steering_angle: 1.58

        # Other
        publish_rate: 50
        enable_odom_tf: true
        pose_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
        twist_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
        cmd_vel_timeout: 5
        base_frame_id: "base_footprint"

        # Geometry (Do not provide the following if you want it to be parsed in the URDF)
        #track: 0.99
        #wheel_steering_y_offset: 0.0
        #wheel_radius: 0.2           
        #wheel_base: 1.02

    # Low-Level Controllers -------------------------------------
    gazebo_ros_control:
        pid_gains:
            steering_fl_joint: {p: 50.0, i: 0.5, d: 5}
            steering_fr_joint: {p: 50.0, i: 0.5, d: 5}
            steering_bl_joint: {p: 50.0, i: 0.5, d: 5}
            steering_br_joint: {p: 50.0, i: 0.5, d: 5}
            wheel_fl_joint: {p: 50.0, i: 2.0, d: 0.001}
            wheel_fr_joint: {p: 50.0, i: 2.0, d: 0.001}
            wheel_bl_joint: {p: 50.0, i: 2.0, d: 0.001}
            wheel_br_joint: {p: 50.0, i: 2.0, d: 0.001}